README.md

```diff
@@ -32,3 +32,383 @@ cd loopback4-example-getting-started && npm install
+application, we need to define our datasource.
+
+Create a new folder in the root directory of the project called `config`,
+and then inside that folder, create a `datasources.json` file. For now, we'll
+be using the memory connector provided with the Juggler.
+
+#### config/datasources.json
+```json
+{
+  "name": "ds",
+  "connector": "memory"
+}
+```
+
+Create another folder called `datasources` in the `src` directory, and inside
+that folder, create a new file called `db.datasource.ts`.
+
+#### src/datasources/db.datasource.ts
+
+```ts
+import * as path from 'path';
+import * as fs from 'fs';
+import { DataSourceConstructor, juggler } from '@loopback/repository';
+
+const dsConfigPath = path.resolve('config', 'datasources.json');
+const config = require(dsConfigPath);
+export const db = new DataSourceConstructor(config);
+```
+
+This will give us a strongly-typed datasource export that we can work with to
+construct our TodoRepository definition.
+
+### Create your repository
+Create another folder in `src` called `repositories` and inside of that folder,
+create two files:
+- `index.ts` (our export helper)
+- `todo.repository.ts`
+
+Our TodoRepository will contain a small base class that uses the
+`DefaultCrudRepository` class from `@loopback/repository` and will define the
+model type we're working with, as well as its ID type. We'll also inject our
+datasource so that this repository can connect to it when executing data
+operations.
+
+#### src/repositories/todo.repository.ts
+```ts
+import { DefaultCrudRepository, DataSourceType } from '@loopback/repository';
+import { Todo } from '../models';
+import { inject } from '@loopback/core';
+
+export class TodoRepository extends DefaultCrudRepository<
+  Todo,
+  typeof Todo.prototype.id
+> {
+  constructor(@inject('datasource') protected datasource: DataSourceType) {
+    super(Todo, datasource);
+  }
+}
+```
+
+
+### Create your controller
+Now, we'll create a controller to handle our Todo routes. Create the
+`src/controllers` directory and two files inside:
+- `index.ts` (export helper)
+- `todo.controller.ts`
+
+In addition to creating the CRUD methods themselves, we'll also be adding
+decorators that setup the routing as well as the expected parameters of
+incoming requests.
+
+#### src/controllers/todo.controller.ts
+```ts
+import {post, param, get, put, patch, del, HttpErrors} from '@loopback/rest';
+import {TodoSchema, Todo} from '../models';
+import {repository} from '@loopback/repository';
+import {TodoRepository} from '../repositories/index';
+
+export class TodoController {
+  constructor(
+    @repository(TodoRepository.name) protected todoRepo: TodoRepository,
+  ) {}
+  @post('/todo')
+  @param.body('todo', TodoSchema)
+  async createTodo(todo: Todo) {
+    if (!todo.title) {
+      return Promise.reject(new HttpErrors.BadRequest('title is required'));
+    }
+    const result = await this.todoRepo.create(todo);
```

Comment:
can we just do `return await this.todoRepo.create(todo);` here?

Reply:
Yep! That actually needs fixing in the real example code as well, though that'll be a different PR.